yamale/yamale.py

```diff
@@ -1,12 +1,20 @@
 #!/usr/bin/env python
 import sys
+import os
 from .schema import Schema
 
 PY2 = sys.version_info[0] == 2
 
 
 def make_schema(path, parser='PyYAML', validators=None):
     # validators = None means use default.
+    # test if the schema.yaml file is empty
+    try:
```

Comment:
You're catching the exception, wrapping it in another exception, and re-raising it again.

Reply:
How do you think you can exit out of such an exception?

Reply:
I'm going to go on vacation for a few days. Take a little time and think how this can be written better. Then we'll discuss when I get back.

Reply:
Np, just for getting things of my head:
- https://docs.python.org/3/tutorial/errors.html#handling-exceptions. Statement after a try catch clause is executed. 
- https://doughellmann.com/blog/2009/06/19/python-exception-handling-techniques/ : Best practices suggest you should raise exception or just write a return statement.
- Multiple test I added checks the functionality across that exception.

Happy reading!

Reply:
Hi, I'm back. Have you thought about a better way to write the following code?
```
    try:
        if not os.path.getsize(path):
            error_message = os.path.basename(path) + ' is an empty file!'
            raise ValueError(error_message)
    except ValueError as e:
        raise ValueError(e)
```

Reply:
@mildebrandt: I don't think you have any better option. There is no existing Python predefined function throwing a `ValueError` checking for the empty file. You require the second raise `ValueError` for exiting out of the execution flow. 

\+ review and/or merge ping.

Reply:
Hi! Sorry for the delay, I was on another vacation. :) It would be better to just remove the `try` block so it's just the following:
```
if not os.path.getsize(path):
    error_message = os.path.basename(path) + ' is an empty file!'
    raise ValueError(error_message)
```

Let me know if you have any questions about that.

Reply:
Hello, Just FYI - In any programming language it is bad practice raising exceptions w/o try catch clause and handling that immediately.